src/main/java/handlers/HandlerPostFileContents.java

```diff
@@ -0,0 +1,57 @@
+package com.td.HttpServer;
+import java.io.IOException;
+public class HandlerPostFileContents implements Ihandler {
+
+  private String path;
+  private byte[] body;
+  private IFileIO fileIO;
+
+  public HandlerPostFileContents(String path, byte[] body, IFileIO fileIO) {
+    this.path = path;
+    this.body = body;
+    this.fileIO = fileIO;
+  }
+
+  public HttpResponse generateResponse() {
+    HttpResponse rtnResponse = new HttpResponse();
+    try {
+      String file = getFileToWrite(path);
+      fileIO.writeContent(file, body);
+      rtnResponse.setResponseCode(201);
+      rtnResponse.addHeader("Location", path);
+    }
+    catch (IOException e) {
+      e.printStackTrace();
+      rtnResponse.setBody("IOException");
+      rtnResponse.setResponseCode(404);
+    }
+    return rtnResponse;
+  }
+
+  private String getFileToWrite(String path) throws IOException {
+    if (!fileIO.exists(path)) {
+      return path;
```

Comment:
Okay. I do see a way to refactor this so that will remove the if check and have only one exit point.